actionpack/test/controller/mime/respond_to_test.rb

```diff
@@ -608,16 +608,20 @@ def test_invalid_format
   end
 
   def test_invalid_variant
+    logger = ActiveSupport::LogSubscriber::TestHelper::MockLogger.new
+    ActionController::Base.logger = logger
```

Comment:
This test overwrites the logger, which will leek into other tests. You have to capture the original value and then set it again after the tests:

```ruby
def test_invalid_variant
  logger = ActiveSupport::LogSubscriber::TestHelper::MockLogger.new
  old_logger, ActionController::Base.logger = ActionController::Base.logger, logger

  # tests...
ensure
  ActionController::Base.logger = old_logger
end
```